Intro to nhst (#441)
arcus/education_modules: intro_to_nhst/intro_to_nhst.md

<!--

author:   Rose Hartman
email:    hartmanr1@chop.edu

> version:  0.0.0

Reviewer: ```suggestion
version:  1.0.0
```

current_version_description: Initial version
module_type: standard

> docs_version: 1.0.0

Reviewer: ```suggestion
docs_version: 1.2.1
```

language: en
narrator: UK English Female
mode: Textbook

title: Introduction to Null Hypothesis Significance Testing

comment:  This is an introduction to NHST for biomedical researchers. 

long_description: Null Hypothesis Significance Testing (NHST) is by far the most commonly used method of statistical inference in research --- regression, ANOVAs, and t-tests are all tests from the NHST framework. This module introduces the important concepts that underlie NHST and prepares you to learn how to use NHST responsibly in your research. It does not assume any prior knowledge of statstics. 

> estimated_time_in_minutes: 

Author: ```suggestion
estimated_time_in_minutes: 40
```

</div>

## The null hypothesis

Often, researchers are curious about an effect, but they usually have only a general scientific hypothesis rather than a precise statistical hypothesis. 

For example, let's say a reseacher is interested in predictors of asthma attacks and has identified an environmental toxin that may be associated with more attacks. 
The researcher's scientific hypothesis would be something like "patients with exposure to X will have more asthma attacks than patients without exposure to X." 
Although that's a perfectly good scientific hypothesis, it's not precise enough to test statistically because "more" could mean any of a range of effects (a slight increase, a huge increase, or anything in between).

<div class = "learn-more">
<b style="color: rgb(var(--color-highlight));">Learning connection</b><br>

For a great discussion of the difference between scientific and statistical hypotheses, see this [post on null hypothesis significance testing](https://education.arcus.chop.edu/null-hypothesis-testing/). 

</div>

> A statistically precise hypothesis would be something like "patients with exposure to X will have an average of 2.6 more asthma attacks per year than patients without exposure to X," but that's not really useful scientifically --- if you end up finding that they have an average of 3.5 more attacks per year instead of 2.6, does that mean your hypothesis was wrong?

Reviewer: ```suggestion
A statistically precise hypothesis would be something like "patients with exposure to X will have an average of 2.6 more asthma attacks per year than patients without exposure to X," but that's not really useful scientifically-- if you end up finding that they have an average of 3.5 more attacks per year instead of 2.6, does that mean your hypothesis was wrong?
```

One way to get around this issue is to **not test your hypothesis of interest itself, but rather its opposite**. 
This is called the **null hypothesis**.

> So you can test the hypothesis that "patients with exposure to X have exactly the same number of asthma attacks as patients without exposure to X" -- that is statistically precise (we're testing that the difference is equal to 0), but it's also still scientifically informative. 

Reviewer: ```suggestion
So you can test the hypothesis that "patients with exposure to X have exactly the same number of asthma attacks as patients without exposure to X"-- that is statistically precise (we're testing that the difference is equal to 0), but it's also still scientifically informative. 
```

Let's return to the example of the asthma researcher studying the difference in number of asthma attacks for patients with and without exposure to toxin X.
Their null hypothesis is that there is no difference in the number of asthma attacks for patients with exposure to X vs. those without exposure to X. 
Let's say they see an average of 5 attacks in the exposure group and 3 in the non-exposure group. 
When they run their statistical test, they'll see one of two things: 

- They get a $p$-value < .05, meaning the observed difference in number of asthma attacks would be very unlikely to occur if there was really no underlying difference between groups based on exposure. In other words, the difference between groups is statistically significant. 
- Or, they get a $p$-value ≥ .05, meaning that if there really is no underlying difference between the groups the observed difference in number of asthma attacks is something that might reasonably have occured just by chance. In other words, the difference between groups is not statistically significant, and we cannot reject the null hypothesis. 

### Type 1 and Type 2 errors

In NHST, there are two different kinds of errors you can have:

> - The first is that you might get a significant result even though actually in reality the null hypothesis is correct --- when we get a $p$-value < .05, the chance of those data happening when the null is true is small (less than 5%), but it's not zero! This is called a **Type 1 error**. You can also think of this as a false positive.

Reviewer: ```suggestion
- The first is that you might get a significant result even though actually in reality the null hypothesis is correct-- when we get a $p$-value < .05, the chance of those data happening when the null is true is small (less than 5%), but it's not zero! This is called a **Type 1 error**. You can also think of this as a false positive.
```

If they found a non-significant difference between groups ($p ≥ .05$), then they would not reject the null hypothesis and that might be a correct decision or it might be a Type 2 error.

### Confidence and power

The probability of a Type 1 error (getting a significant result when the null hypothesis is in fact true) is called **confidence**, and it's actually the alpha critierion we set. 

As a reminder, a $p$-value is the probability of the effect observed in the data (or larger effect) if the null hypothesis were indeed true. 
And Type 1 error is when the null is true but you reject the null hypothesis, which we do whenever $p$ is less than alpha (typically .05). 
So by definition, we're saying we'll reject true null hypotheses 5% of the time. 

<div class = "behind-the-scenes">
<b style="color: rgb(var(--color-highlight));">Behind the scenes</b><br>

**Why .05?**

There's no good reason alpha should be .05, it's just a convention we've settled on. 
It could just as easily be .01 or .1, or even .5 (although that would mean we'd be accepting 50% chance of Type 1 errors, which would add a lot of noise to the published literature!). 

</div>

> The probability of a Type 2 error (failing to reject the null hypothesis when it's really false) is represented by the letter beta. 

Reviewer: ```suggestion
The probability of a Type 2 error (failing to reject the null hypothesis when it's really false) is represented by the letter **beta**. 
```

> Because there are only two possible outcomes when the null is false (either correctly rejecting it or having a Type 2 error), the probability of correctly rejecting the null hypothesis (getting a significant result when the null is truly false) 1-beta, and it is called **power**.

Reviewer: ```suggestion
Because there are only two possible outcomes when the null is false (either correctly rejecting it or having a Type 2 error), the probability of correctly rejecting the null hypothesis (getting a significant result when the null is truly false) is 1-beta, and it is called **power**.
```

To read more about power analyses, check out the article [Power and sample size](https://pubmed.ncbi.nlm.nih.gov/18450060) (Case & Ambrosius, 2007).

</div>

Especially in fields where data collection is expensive or difficult, researchers may be working with small sample sizes; if you're trying to test for a subtle effect with a small sample, then even if the effect is there your power might be too low for you to expect to get a significant result.
A power analysis can make it clear to you that, with the sample size you're planning, it may not be worth your time to test for the effect you're interested in. 
If a study doesn't have "enough" power (often 80% is used as a cutoff), it is said to be **underpowered**.

<div class = "learn-more">
<b style="color: rgb(var(--color-highlight));">Learning connection</b><br>

For a discussion of the prevalence and ramifications of underpowered studies in a particular field, see [Power failure: why small sample size undermines the reliability of neuroscience](https://www.nature.com/articles/nrn3475) (Button et al., 2013).

Spoilers: They estimate the median power of studies in the neurosciences to be between 8% and 31%! 

> In other words, assuming that the effect they're testing for are real (the null hypotheses are false) most of these studies had less than a 50% chance of getting a significant result.

Reviewer: ```suggestion
In other words, assuming that the effects they're testing for are real (the null hypotheses are false) most of these studies had less than a 50% chance of getting a significant result.
```

</div>

## The Bayesian approach

NHST is not the only way to infer information about a population from a sample!

Another popular approach is Bayesian inference, named for [Thomas Bayes](https://en.wikipedia.org/wiki/Thomas_Bayes), who forumulated the [theorem](https://en.wikipedia.org/wiki/Bayes%27_theorem) on which it is based. 
Briefly, Bayes' theorem tells us how to update the probability of something given new evidence. 

<div class = "learn-more">
<b style="color: rgb(var(--color-highlight));">Learning connection</b><br>

For a much more thorough (but approachable!) explanation of Bayes' theorem, see [the StatQuest video, "Bayes' Theorem, Clearly Explained"](https://statquest.org/bayes-theorem-clearly-explained/).

</div>

For example, without any more specific information, you would assume a given patient's probability of having beast cancer is about the same as the population probability of that cancer. 

> If you find out that she's had a mamogram that detected cancer (new evidence), how does that change your estimation of her probability of having breast cancer? 

Reviewer: ```suggestion
If you find out that she's had a mammogram that detected cancer (new evidence), how does that change your estimation of her probability of having breast cancer? 
```

> It still wouldn't be 100% (mamograms have false positives), but it would be higher than before you knew about the mamogram. 

Reviewer: ```suggestion
It still wouldn't be 100% (mammograms have false positives), but it would be higher than before you knew about the mammogram. 
```

> Bayes' theorem expresses precisely, in math, what the updated probability would be, given the prior probability and the strength of the evidence (how good of a test is a mamogram?). 

Reviewer: ```suggestion
Bayes' theorem expresses precisely, in math, what the updated probability would be, given the prior probability and the strength of the evidence (how good of a test is a mammogram?). 
```

In a research context, you would specify a prior probability for your hypothesis (if you want to be conservative, you might set it to zero, like a null hypothesis). 
Then you collect your data and estimate the strength of the evidence, for example as an effect size, and you can use Bayes' theorem to calculate what the probability of your hypothesis is now, given the evidence. 

**In other words, Bayesian inference allows to you directly test the questions you're actually curious about in a way that is much more intutive than NHST.**

> Bayesian inference also provides a natural way to incoporate additional knowledge into your hypothesis test --- if you're testing a hypothesis that seems very unlikely, the evidence should need to be much stronger than if you're testing for something that's pretty common sense. 

Reviewer: ```suggestion
Bayesian inference also provides a natural way to incoporate additional knowledge into your hypothesis test-- if you're testing a hypothesis that seems very unlikely, the evidence should need to be much stronger than if you're testing for something that's pretty common sense. 
```

For example, there is a real, published study in a mainstream journal purporting to show evidence that research participants have some abililty to sense the future before it happens ([Bem, 2011](https://pubmed.ncbi.nlm.nih.gov/21280961/)). 

> That article triggered several responses from other scientists arguing about a number of details in the statistical methods employed in the original study, including the fact that a Bayesian analysis would have allowed the analyst to specify a prior probability for being able to correctly anticipate random images, and that prior probability should have been quite low given the very reasonable skepticsm around ESP (for example, see [Rouder & Morey, 2011](https://link.springer.com/article/10.3758/s13423-011-0088-7)). 

Reviewer: ```suggestion
That article triggered several responses from other scientists arguing about a number of details in the statistical methods employed in the original study, including the fact that a Bayesian analysis would have allowed the analyst to specify a prior probability for being able to correctly anticipate random images, and that prior probability should have been quite low given the very reasonable skepticsm around extrasensory perception, or ESP (for example, see [Rouder & Morey, 2011](https://link.springer.com/article/10.3758/s13423-011-0088-7)). 
```

Reviewer: This is just a suggestion, feel free to take or leave, but there could be some folks who aren't familiar with what ESP refers to.

<h3>So why aren't we all just using Bayesian inference instead of NHST?</h3>

Well, some people are! 
Bayesian methods are becoming more common, especially as more statistical software supports them.

But there are a few reasons we're opting to teach mostly NHST here instead of Bayesian inference:

- There's a huge body of existing literature using NHST, so it's important you understand those methods even if you prefer Bayesian inference for your own analysis.
- The available tools are still much more user friendly for NHST. 

> - In a practical sense, using NHST vs. Bayesian inference usually won't change your conclusions. If you analyze your data with Bayesian methods and then re-analyze it with NHST, you'll almost always see comprable results. And that makes sense --- it's the same data, after all! 

Reviewer: ```suggestion
- In a practical sense, using NHST vs. Bayesian inference usually won't change your conclusions. If you analyze your data with Bayesian methods and then re-analyze it with NHST, you'll almost always see comparable results. And that makes sense-- it's the same data, after all! 
```

## The dangers of p-hacking

As we mentioned earlier, the Type 1 error rate (the probability of incorrectly rejecting the null hypothesis when it is in fact true) is set by convention at 5%. 

> That suggests that, if you're scientififc hypothesis is wrong and you're testing for an effect that isn't really there (i.e. the null hypothesis is true), the chance of you getting a significant result is pretty small: 1 in 20. 

Reviewer: ```suggestion
That suggests that, if your scientific hypothesis is wrong and you're testing for an effect that isn't really there (i.e. the null hypothesis is true), the chance of you getting a significant result is pretty small: 1 in 20. 
```

In fact, though, the way many people routinely analyze data results in an actual Type 1 error rate that is much, much higher. 

> Here are a few common practices that are actually p-hacking: 

Reviewer: ```suggestion
Here are a few common practices that are actually **p-hacking**: 
```

> - Use NHST to run a test that didn't occur to you until you started looking at the data

Reviewer: ```suggestion
- Using NHST to run a test that didn't occur to you until you started looking at the data
```

You can do this privately, just for your own benefit, or you can publish your pregistration publicly on a repository like the [Open Science Foundation](https://help.osf.io/article/330-welcome-to-registrations).

<div class = "learn-more">
<b style="color: rgb(var(--color-highlight));">Learning connection</b><br>

To read more about the benefits of preregistration, read the *PLOS Computational Biology* Methods paper [Ten simple rules for socially responsible science](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010954#sec004) (Zivony, Kardosh, Timmins & Reggev, 2023). 

</div>

## Quiz

True or False: NHST is based on the assumption that your sample is drawn **randomly** from your population of interest.

[(X)] True
[( )] False
***
<div class = "answer">

Yes, NHST assumes you have a random sample from your population of interest. 
It's often not feasible to obtain a true random sample from your population of interest, though, so part of interpreting your results is considering how your sample may differ from the population you want to generalize your results to.
This is often an important part of the discussion section in a published paper.

</div>
***

> Consider a researher who has been developing a checklist to improve care for postpartum people. The researcher has the opportunity to conduct a randomized trial at their hospital, so they randomly assign one group of birthing patients to be treated with the checklist protocol and another to get care as usual. They hypothesize that outcomes will be better in the group that gets the checklist. What null hypothesis should they test? 

Reviewer: ```suggestion
Consider a researcher who has been developing a checklist to improve care for postpartum people. The researcher has the opportunity to conduct a randomized trial at their hospital, so they randomly assign one group of birthing patients to be treated with the checklist protocol and another to get care as usual. They hypothesize that outcomes will be better in the group that gets the checklist. What null hypothesis should they test? 
```

[[ that there is no difference in outcomes for the group that gets the checklist vs. the group that gets care as usual]]
<script>
  let input = "@input".trim();
  /.*/i.test(input);
</script>
***
<div class = "answer">

(Note that we can't automatically grade an open-ended question like this, so it will be marked "correct" no matter what you write.)

The null hypothesis would be that there is no difference in outcomes for the group that gets the checklist vs. the group that gets care as usual.

</div>
***

Which of the following are true statements about the null hypothesis? Select all that apply.

[[ ]] If your results are not significant ($p ≥ .05$), you conclude that the null hypothesis is true.

> [[X]] If your results are significant ($p ≥ .05$), you conclude that the null hypothesis is false.

Reviewer: ```suggestion
[[X]] If your results are significant ($p < .05$), you conclude that the null hypothesis is false.
```
